### PR TITLE
Correctly deal with sensor rotation in AHRS

### PR DESCRIPTION
**Pull request by dellaert**
Sensor ~displacement~ rotation was not handled correctly in the AHRS factors. I spent a day tracking it down and fixed it:

- Significantly cleaned up `testAHRSFactor.cpp`, now uses `FactorTesting.h` macros
- To try and debug, deprecated two methods in `PreintegratedRotation` with hard-to-understand Jacobians
- Instead, added a function object with sane Jacobians, and fixed bug in Jacobians (missing chain rule).
- End-result is still not exactly how I'd like it ("insane" Jacobian in `biascorrectedDeltaRij`) but fixes bug

Tested:
- add `testPreintegratedRotation.cpp`, all tests pass
- Added new end-to-end integration test in `testAHRSFactor.cpp`, inspired by `testImuFactor.cpp`
- added a test in `testManifoldPreintegration.cpp` to establish equivalence
- added tests on deprecated methods, work, but deleted them in commit 1ac286f97a6fb482c26d4a9c9d042c4fd065e74d

### CODE REVIEW

File: gtsam/navigation/PreintegratedRotation.cpp
Comment (dellaert) [inline gtsam/navigation/PreintegratedRotation.cpp:93]: This is the bugfix!

File: gtsam/navigation/tests/testAHRSFactor.cpp
Comment (dellaert) [inline gtsam/navigation/tests/testAHRSFactor.cpp:410]: This is the new end-to-end test, with sensor rotation and large bias. No aiding, so prior on bias makes solution observable.

File: gtsam/navigation/PreintegratedRotation.h
Comment (varunagrawal) [inline gtsam/navigation/PreintegratedRotation.h]: Specify that this is an "optional" jacobian?

File: gtsam/navigation/PreintegratedRotation.h
Comment (varunagrawal) [inline gtsam/navigation/PreintegratedRotation.h]: Same here, specifying that `H` is an optional Jacobian.

File: gtsam/navigation/PreintegratedRotation.h
Comment (varunagrawal) [inline gtsam/navigation/PreintegratedRotation.h:166]: Awesome!

File: gtsam/navigation/PreintegratedRotation.h
Comment (varunagrawal) [inline gtsam/navigation/PreintegratedRotation.h:41]: I am not sure I understand the point of making this a struct since the members here don't seem to be used beyond the `operator()` function call.
The biggest question I have is how would this affect inlining of the function? In terms of keyboard-typing, there is no benefit since we have to pass in the same number of variable (over 2 lines now instead of 1) so if this makes it hard for certain compilers to inline the function, then it may not be the best option.

**Comment by dellaert** [comment]
@andre-michelin Please review and confirm these changes work in our code?

File: gtsam/navigation/PreintegratedRotation.h
Comment (dellaert) [reply gtsam/navigation/PreintegratedRotation.h:41]: I opted to make it very clear that this is a function of the bias and all other arguments are parameters. I acknowledge it might affect inlining, although I have no idea how so I'm opting for clarity here.

File: gtsam/navigation/PreintegratedRotation.h
Comment (dellaert) [reply gtsam/navigation/PreintegratedRotation.h]: Done

File: gtsam/navigation/PreintegratedRotation.h
Comment (dellaert) [reply gtsam/navigation/PreintegratedRotation.h]: Done

**Comment by dellaert** [comment]
Thanks @varunagrawal , I addressed two of your comments in my local repo but will wait for Andre's comments to push, so we have to do CI only once more before merging.

File: gtsam/navigation/tests/testAHRSFactor.cpp
Comment (andre-michelin) [inline gtsam/navigation/tests/testAHRSFactor.cpp:470]: typo

**Review by andre-michelin — APPROVED** [review]

**Pull request merged**
